extensions/amp-list/0.1/amp-list.js

```diff
@@ -48,10 +54,35 @@ export class AmpList extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    return this.populateList_();
+  }
+
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const srcMutation = mutations['src'];
+    if (srcMutation != undefined) {
+      const p = this.populateList_();
+      if (getMode().test) {
+        this.mutationPromise_ = p;
```

Comment:
Unit tests accessing private properties/methods is definitely fine, I prefer that over exporting specific methods/properties just for testing.